### PR TITLE
feat: extract custom domain utils to a service

### PR DESCRIPTION
**Pull request by cquintana92**
This PR performs the following changes:

1. Offer a new function that loads a dictionary from an env var
2. Allow to define partner domains (not in use yet)
3. Extracting logic from the controller into a service so it can be reused
4. Move the DMARC record to `app.constants`

### CODE REVIEW

File: app/config.py [hunk 1]
```diff
@@ -35,6 +35,34 @@ def sl_getenv(env_var: str, default_factory: Callable = None):
...
+
+    components = value.split(";")
+    result = {}
+    for component in components:
+        if component == "":
+            continue
+        parts = component.split("=")
+        if len(parts) != 2:
+            raise Exception(f"Invalid config for env var {env_var}")
+        result[parts[0]] = parts[1]
```
Comment (acasajus) [inline app/config.py:61]:
```suggestion
        result[parts[0].strip()] = parts[1].strip()
```

File: app/config.py [hunk 2]
```diff
@@ -35,6 +35,34 @@ def sl_getenv(env_var: str, default_factory: Callable = None):
     return literal_eval(value)
 
 
+def get_env_dict(env_var: str, default_factory: Callable = None) -> dict[str, str]:
```
Comment (acasajus) [inline app/config.py:38]: Worth leaving default as empty dict directly?

Comment (cquintana92) [reply app/config.py:38]: Or even better, not allowing to pass it as it will always be a dict

Comment (acasajus) [reply app/config.py]: 👍

File: app/config.py (hunk 1 above)
Comment (cquintana92) [reply app/config.py]: Changed

**Review by acasajus — APPROVED** [review]

**Pull request merged**